packages/api-client/src/v2/features/modal/hooks/use-modal-sidebar.ts

```diff
@@ -0,0 +1,212 @@
+import type { HttpMethod } from '@scalar/helpers/http/http-methods'
+import { createSidebarState, generateReverseIndex, getChildEntry } from '@scalar/sidebar'
+import type { WorkspaceStore } from '@scalar/workspace-store/client'
+import { getParentEntry } from '@scalar/workspace-store/navigation'
+import type { TraversedEntry } from '@scalar/workspace-store/schemas/navigation'
+import { type MaybeRefOrGetter, computed, toValue, watch } from 'vue'
+
+import type { RoutePayload } from '@/v2/features/modal/helpers/resolve-route-parameters'
+import { generateLocationId } from '@/v2/helpers/generate-location-id'
+
+export type UseModalSidebarReturn = {
+  handleSelectItem: (id: string) => void
+  state: ReturnType<typeof createSidebarState<TraversedEntry>>
+  getEntryByLocation: (location: {
+    document: string
+    path?: string
+    method?: HttpMethod
+    example?: string
+  }) => TraversedEntry | undefined
+}
+
+/**
+ * useSidebarState - Custom hook to manage the sidebar state and navigation logic in the Scalar API client
+ *
+ * This composable manages the sidebar structure, synchronizes selection state
+ * with the current route, and provides a handler for selecting sidebar items.
+ *
+ * Example usage:
+ *
+ * const { handleSelectItem, sidebarState } = useSidebarState({
+ *   workspaceStore,
+ *   workspaceSlug,
+ *   documentSlug,
+ *   path,
+ *   method,
+ *   exampleName,
+ * })
+ */
+export const useModalSidebar = ({
+  workspaceStore,
+  documentSlug,
+  path,
+  method,
+  exampleName,
+  route,
+}: {
+  workspaceStore: MaybeRefOrGetter<WorkspaceStore | null>
+  documentSlug: MaybeRefOrGetter<string | undefined>
+  path: MaybeRefOrGetter<string | undefined>
+  method: MaybeRefOrGetter<HttpMethod | undefined>
+  exampleName: MaybeRefOrGetter<string | undefined>
+  route: (payload: RoutePayload) => void
```

Comment:
```suggestion
  workspaceStore: Computed<WorkspaceStore | null>
  documentSlug: Computed<string | undefined>
  path: Computed<string | undefined>
  method: Computed<HttpMethod | undefined>
  exampleName: Computed<string | undefined>
  route: (payload: RoutePayload) => void
```
we are passing in computed an its only used in one place, no need for the ambiguity of maybeRefOrGetter